### PR TITLE
Validate url by using newspaper3k's is_valid_url()

### PR DESCRIPTION
**Pull request by sungho-cho**
newspaper3k's `is_valid_body()` is too strict to validate most URLs.
Even CNN articles fail to pass the [function](https://github.com/codelucas/newspaper/blob/master/newspaper/article.py#L322).

Along with `is_valid_body()`, we should also use `is_valid_url()` which has the main logic [here](https://github.com/codelucas/newspaper/blob/master/newspaper/urls.py#L102).

It has a fairly reasonable article validation process which includes:
- Basic URL regex check
- Check if a web page contains a date
- Look for certain keywords (e.g. story, article, news, press, etc.)

To make it less strict, we now check both `is_valid_body()` and `is_valid_url()` and also the length of the body text.

### CODE REVIEW

File: aws/lambda/lambda_function.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def handler(event, context):
     logging.debug(f"Article authors: {article.authors}")
     
     # Send error response if the given URL does not contain a valid article
-    # https://github.com/codelucas/newspaper/blob/master/newspaper/article.py#L322
-    if not article.is_valid_body():
+    # https://github.com/codelucas/newspaper/blob/master/newspaper/urls.py#L102
+    if not article.is_valid_url():
```
Comment (junhur) [inline aws/lambda/lambda_function.py]: Just curious, does this work for Medium articles as well? 
Many medium article links are formatted `/@username/title-of-the-article`, so was curious if the date check will still work.

File: aws/lambda/lambda_function.py [hunk 2]
```diff
@@ -29,8 +29,8 @@ def handler(event, context):
     logging.debug(f"Article authors: {article.authors}")
     
     # Send error response if the given URL does not contain a valid article
-    # https://github.com/codelucas/newspaper/blob/master/newspaper/article.py#L322
-    if not article.is_valid_body():
```
Comment (junhur) [inline aws/lambda/lambda_function.py:33]: How about we keep this `is_valid_body()` check and only perform an additional `is_valid_url()` for cases that fail the first check?

**Review by junhur — COMMENTED** [review]
Thanks for getting to this really quick!
Left some thoughts 👍

File: aws/lambda/lambda_function.py (hunk 1 above)
Comment (sungho-cho) [reply aws/lambda/lambda_function.py]: Some Medium articles like [this](https://levelup.gitconnected.com/how-i-got-in-to-amazon-microsoft-google-all-from-studying-these-resources-31724508ce0e) passes `is_valid_url`.

Some fail, however, like [this one](https://medium.com/double-pointer/guide-to-bloom-filters-50475843dfa2).

I'm now browsing other ways to validate articles...

**Review by chorongi — COMMENTED** [review]
If is_valid_body or is_valid_url is false, what does field article.text have? Wanted to make sure when it has valid text

**Comment by sungho-cho** [comment]
@chorongi @junhur 
This is what `article.text` looks like when the articles has invalid url and body.
image

File: aws/lambda/lambda_function.py (hunk 2 above)
Comment (sungho-cho) [reply aws/lambda/lambda_function.py:33]: I made the condition a little more lenient

**Review by chorongi — APPROVED** [review]
The more linient condition looks better! I think we can further play around with our tool to test how its effective for a diverse set of articles. Thank you for handling this 👍

**Pull request merged**